ext_mongodb.php

```diff
@@ -929,21 +929,21 @@ public function __construct(int $readPreference, array $tagSets = null, array $o
 			$this->_setReadPreferenceTags( $newTagSets );
 		}
 
-		if ( array_key_exists( 'maxStalenessMS', $options ) )
+		if ( array_key_exists( 'maxStalenessSeconds', $options ) )
 		{
-			$maxStalenessMS = (int) $options['maxStalenessMS'];
+			$maxStalenessSeconds = (int) $options['maxStalenessSeconds'];
 
-			if ( $maxStalenessMS < 0 )
+			if ( $maxStalenessSeconds < 0 )
 			{
-				Utils::throwHippoException( Utils::ERROR_INVALID_ARGUMENT, "Expected maxStalenessMS to be >= 0, {$maxStalenessMS} given" );
+				Utils::throwHippoException( Utils::ERROR_INVALID_ARGUMENT, "Expected maxStalenessSeconds to be >= 0, {$maxStalenessSeconds} given" );
```

Comment:
Based on the CDRIVER-1933 code review (ongoing), `MONGOC_SMALLEST_MAX_STALENESS_SECONDS` is a private constant and is only checked at runtime when executing an operation. The C driver performs no validation during URI parsing or construction of a read preference. Read preference validation only requires it be a positive number (> 0). The spec also talks about allowing `-1` to specify the default, so we need to make an exception for that here (no pun intended).

I think the correct behavior is to allow -1 and leave the default in place and throw otherwise if the value is <= 0 and state that it must be greater than 0.